### PR TITLE
Update implicit-conversions.md

### PR DESCRIPTION
**Pull request by oliver-brm**
I'm not sure if it is intentional, but the Tour-of-Scala documentation does not mention how to bring implicit conversions into scope. In fact, that information is a bit hard to find anywhere else, so I thought I'll propose to add it here?

### CODE REVIEW

File: _tour/implicit-conversions.md [hunk 1]
```diff
@@ -38,6 +38,45 @@ In the second case, a conversion `c` is searched for, which is applicable to `e`
...
+  implicit def fromStringToStudent(name: String): Student = Student(name)
+}
+
+import ImplicitConversions._
+object Usage:
+  def main(args: Array[String]) = "Reginald".printName
+```
+{% endtab %}
+{% tab 'Scala 3' %}
+In Scala 3, an implicit conversion must be explicitly imported in order to be applicable.
```
Comment (bishabosha) [inline _tour/implicit-conversions.md:62]: this isn't true, you can also have a given import:
```scala
import Conversions.given
import Conversions.{given Conversion[String, Student]}
```

Comment (bishabosha) [reply _tour/implicit-conversions.md:62]: or if the implicit conversion is in the companion object of the argument type, (e.g. String in this case) then no import is necessary

File: _tour/implicit-conversions.md [hunk 2]
```diff
@@ -38,6 +38,45 @@ In the second case, a conversion `c` is searched for, which is applicable to `e`
 
 An example is to compare two strings `"foo" < "bar"`. In this case, `String` has no member `<`, so the implicit conversion `Predef.augmentString("foo") < "bar"` is inserted. (`scala.Predef` is automatically imported into all Scala programs.)
 
+### How are implicit conversions brought into scope? ###
+
+{% tabs implicit-conversion-scope class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+In Scala 2, an implicit convesion is brought into scope by importing from `ImplicitConversions`.
```
Comment (bishabosha) [inline _tour/implicit-conversions.md:45]:
```suggestion
In Scala 2, an implicit conversion is brought into scope by importing from the scope it was defined in, (e.g. `Conversions` in this case).
```

File: _tour/implicit-conversions.md (hunk 1 above)
Comment (oliver-brm) [reply _tour/implicit-conversions.md]: Thank you for all the suggestions! I've learned a lot! I've also improved the documentation.

File: _tour/implicit-conversions.md [hunk 3]
```diff
@@ -38,6 +38,60 @@ In the second case, a conversion `c` is searched for, which is applicable to `e`
...
+object Student:
+  given Conversion[Student, Int] = _.name.length
+  
+object Conversions:
+  given fromStringToStudent: Conversion[String, Student] = Student(_)
+```
+
+The following imports would bring the `Conversion[String, Student]` into scope:
+  - `import Conversions.given`
+  - `import Conversions.{given[String, Student]}`
```
Comment (bishabosha) [inline _tour/implicit-conversions.md:80]:
```suggestion
  - `import Conversions.{given Conversion[String, Student]}`
```

File: _tour/implicit-conversions.md [hunk 4]
```diff
@@ -38,6 +38,60 @@ In the second case, a conversion `c` is searched for, which is applicable to `e`
...
+  - `import Conversions.fromStringToStudent`
+
+If the implicit conversion is in the companion object of the argument type, (e.g. `Student` in this case), then no import is necessary.
```
Comment (bishabosha) [inline _tour/implicit-conversions.md:83]: this also applies to Scala 2

Comment (oliver-brm) [reply _tour/implicit-conversions.md:88]: Thanks, I added it to the Scala 2 example as well.

File: _tour/implicit-conversions.md (hunk 1 above)
Comment (oliver-brm) [reply _tour/implicit-conversions.md]: Just asking: There's no way to add a companion object to a Java class like `java.lang.String`, is it? I tried that but either the compiler runs into trouble or it does not find the conversion.

**Review by bishabosha — APPROVED** [review]
very nice, thanks for contributing!

File: _tour/implicit-conversions.md (hunk 1 above)
Comment (bishabosha) [reply _tour/implicit-conversions.md]: companion objects must be defined in the same source file :)

File: _tour/implicit-conversions.md [hunk 5]
```diff
@@ -38,6 +38,65 @@ In the second case, a conversion `c` is searched for, which is applicable to `e`
...
+```
+
+The following imports would bring the `Conversion[String, Student]` into scope:
+  - `import Conversions.given`
+  - `import Conversions.{given Conversion[String, Student]}`
+  - `import Conversions.fromStringToStudent`
+
+If the implicit conversion is in the companion object of the argument type, (e.g. `Student` in this case), then no import is necessary.
+
+```scala mdoc
```
Comment (bishabosha) [inline _tour/implicit-conversions.md:90]:
```suggestion
```scala
```

**Comment by bishabosha** [comment]
I've just remembered that you wrote all this when the next section "How are implicit conversions selected" is supposed to answer this point, by linking to the scala FAQ, maybe it would be good to merge the sections

File: _tour/implicit-conversions.md [hunk 6]
```diff
@@ -38,6 +38,65 @@ In the second case, a conversion `c` is searched for, which is applicable to `e`
...
 An example is to compare two strings `"foo" < "bar"`. In this case, `String` has no member `<`, so the implicit conversion `Predef.augmentString("foo") < "bar"` is inserted. (`scala.Predef` is automatically imported into all Scala programs.)
 
+### How are implicit conversions brought into scope? ###
+
+{% tabs implicit-conversion-scope class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+In Scala 2, an implicit conversion is brought into scope by importing from the object that defined it, (e.g. `Conversions` in this case). If the implicit conversion is in the companion object of the argument type, (e.g. `Student` in this case), then no import is necessary.
+
+```scala mdoc
+case class Student(name: String)
```
Comment (bishabosha) [inline _tour/implicit-conversions.md:48]:
```suggestion
import scala.language.implicitConversions

case class Student(name: String)
```

**Comment by oliver-brm** [comment]
> I've just remembered that you wrote all this when the next section "How are implicit conversions selected" is supposed to answer this point, by linking to the scala FAQ, maybe it would be good to merge the sections

OK, I can do this. The FAQ, however, only refers to this [StackOverflow](https://stackoverflow.com/questions/5598085/where-does-scala-look-for-implicits/5598107#5598107) article, written back in 2011. I personally would not link to it from Tour of Scala, but I can place it under a _Further reading_ section. WDYT?

**Review by bishabosha — APPROVED** [review]
Thanks for being patient :)

**Comment by oliver-brm** [comment]
> Thanks for being patient :)

Thanks for being patient, too! I’m happy to have contributed. 😊

**Pull request merged**